core/base/src/TBuffer.cxx

```diff
@@ -23,6 +23,17 @@ const Int_t  kExtraSpace        = 8;   // extra space at end of buffer (used for
 
 ClassImp(TBuffer);
 
+namespace ROOT {
+namespace Internal {
+   /// Default streamer implementation used by ClassDefInline to avoid
+   /// requirement to include TBuffer.h
+   void DefaultStreamer(TBuffer& R__b, const TClass *cl, void *objpointer) {
```

Comment:
If you rewrite this to `ROOT::Internal::DefaultStreamer()` you'll get an error if the declared and defined signatures don't match. As you write it here you won't. See https://godbolt.org/g/Lp4Qoh